plugins/woocommerce/src/Caches/OrderAggregateCache.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Caches;
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore;
+
+/**
+ * A class to cache aggregates from orders.
+ */
+class OrderAggregateCache {
+
+	/**
+	 * Option name.
+	 */
+	const OPTION_NAME = 'woocommerce_orders_aggregate_cache';
+
+	/**
+	 * The running totals for all order statuses.
+	 *
+	 * @var array|null
+	 */
+	private $totals = null;
+
+	/**
+	 * Whether or not the count has changed since loading.
+	 *
+	 * @var bool
+	 */
+	private $has_changed = false;
+
+	/**
+	 * Get the aggregate by identifier.
+	 *
+	 * @param string|string[] $status The order status to count.
+	 * @return int
+	 * @throws \Exception Exception thrown if status it not valid.
+	 */
+	public function get_count( $status ): int {
+		$status         = (array) $status;
+		$valid_statuses = $this->get_valid_statuses();
+
+		if ( ! empty( array_diff( $status, $valid_statuses ) ) ) {
+			throw new \Exception();
+		}
+
+		if ( ! $this->totals ) {
+			$cache        = get_option( self::OPTION_NAME );
+			$this->totals = $cache ? $cache : $this->count_orders_by_status();
+		}
+
+		return array_sum( array_intersect_key( $this->totals, array_flip( $status ) ) );
```

Comment:
`array_flip( $status )` doesn't seem right. When it comes to multiple statuses passed in, I believe we aim for the result to be filled with 0-values.

Reply:
This is just using the existing count logic ([here](https://github.com/woocommerce/woocommerce/blob/c1ec710de807934d06eda88109aab4cc50d5155f/plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php#L619)).

I haven't spent a lot of time analyzing this, but is this returning invalid values for you?

Reply:
Thanks for looking into it!

It was confusing to read, but [apparently](https://3v4l.org/e964i) is working as intended %)